src/ipc/ipc4/handler.c

```diff
@@ -572,6 +583,10 @@ static int ipc4_init_module_instance(struct ipc4_message_request *ipc4)
 	tr_dbg(&ipc_tr, "ipc4_init_module_instance %x : %x", (uint32_t)module.primary.r.module_id,
 	       (uint32_t)module.primary.r.instance_id);
 
+	/* Pass IPC to target core */
+	if (!cpu_is_me(module.extension.r.core_id))
+		return ipc_process_on_core(module.extension.r.core_id, false);
```

Comment:
Initially I was thinking this is out of scope of this PR, as ipc_process_on_core with POSIX error codes was there already and I'm using it the same way.
But after your comment I agree it's better to fix this. How about wrapper like `ipc4_process_on_core() ` which would translate error codes?

Reply:
sure, a wrapper should work

Reply:
strange, I thought I replied already... So maybe this will be duplicated, sorry. Yes, @pblaszko you're right - there is already one case of returning the error code from `ipc_process_on_core()` in `set_pipeline_state()` there, it has to be fixed too. Since this your PR is adding multicore support, I think it would be ok to add that fix as the first commit to this PR.